### PR TITLE
Impro1347 spot attributes

### PR DESCRIPTION
**Pull request by cgsandford**
Puts in place functionality to update the spot cube title attribute via command line argument, which is done in associated suite PR.  Removes "amend_attributes" functionality from CLI as this should never be used - the spot extraction subroutine inherits from the grid, and only the title is updated.

This PR also removes some utilities I added earlier in this work, before we decided on a more flexible approach.

Testing:
 - [x] Ran tests and they passed OK
 - [x] Added new tests for the new feature(s)

### CODE REVIEW

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/spotdata/spot_extraction.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/metoppv_tech/improver/pullRequest?prid=4466200&bid=15083499)

File: lib/improver/cli/spot_extract.py [hunk 1]
```diff
@@ -78,6 +77,9 @@ def main(argv=None):
                         " site-altitude.")
     parser.add_argument("output_filepath", metavar="OUTPUT_FILEPATH",
                         help="The output path for the resulting NetCDF")
+    parser.add_argument("--new_title", metavar="NEW_TITLE", default=None,
+                        help="Title attribute for spot-extracted data. If not "
+                        "set, this attribute is removed.", type=str)
```
[the remark this answers is not in the document]
Comment (cgsandford) [reply lib/improver/cli/spot_extract.py]: Will add something to help string - basically "title is not prescriptive in CF standard, so may include grid information which is no longer correct after spot-extraction".

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/spotdata/spot_extraction.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/improver/improver/pullRequest?prid=4466200&bid=15083499)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/spotdata/spot_extraction.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/improver/improver/pullRequest?prid=4466200&bid=15083499)

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/spotdata/spot_extraction.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/improver/improver/pullRequest?prid=4466200&bid=15083499)

File: lib/improver/spotdata/spot_extraction.py [hunk 2]
```diff
@@ -168,6 +168,9 @@ def process(self, neighbour_cube, diagnostic_cube):
                 their grid point neighbours.
             diagnostic_cube (iris.cube.Cube):
                 A cube of diagnostic data from which spot data is being taken.
+            new_title (str or None):
+                New title for spot-extracted data.  If None, this attribute is
+                removed from the output cube.
```
Comment (MoseleyS) [inline lib/improver/spotdata/spot_extraction.py]: Please can you add the extended help-string here too.

**Comment by benfitzpatrick** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/improver/spotdata/spot_extraction.py  1
         

Complexity decreasing per file
==============================
+ lib/improver/cli/spot_extract.py  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/improver/improver/pullRequest?prid=4466200&bid=15083499)

**Comment by bayliffe** [comment]
The real time suite has been broken by our metadata changes. If we can expedite this PR and the subsequent suite PR the issue should be resolved. @MoseleyS  could you please check you are happy with the changes ASAP and approve this if so.

**Pull request merged**